tests/raftstore/test_snap.rs

```diff
@@ -288,7 +288,7 @@ struct StaleSnap {
 }
 
 impl Filter<RaftMessage> for Arc<StaleSnap> {
-    fn before(&self, msgs: &mut Vec<RaftMessage>) {
+    fn before(&self, msgs: &mut Vec<RaftMessage>) -> FilterResult {
```

Comment:
we can return Result<()> directly and no need to check msg.empty() later. 
In the delay filter, we can return ok, and for other filters, we can use a function like 
```
fn check_droppable_err(msg:&Vec<RaftMessage>) -> Result<()> {
   if msg.empty() {
       return DropableError
   } else {
       return ok
  }
}
```